### PR TITLE
Fix CeedOperatorFlags

### PR DESCRIPTION
**Pull request by jeremylt**
Updated `CeedOperator` flags to be flags and not `NULL`.

### CODE REVIEW

File: backends/occa/ceed-occa-operator.c [hunk 1]
```diff
@@ -331,45 +331,52 @@ static int CeedOperatorApply_Occa(CeedOperator op,
...
     dbg("\n[CeedOperator][Apply] %d/%d Input Evecs:",i,qf->numinputfields-1);
-    // Restriction
-    if (op->inputfields[i].Erestrict) {
+    // No Restriction
+    if (op->inputfields[i].Erestrict == CEED_RESTRICTION_IDENTITY) {
+      CeedEvalMode emode = qf->inputfields[i].emode;
+      if (emode & CEED_EVAL_WEIGHT) {
+        dbg("[CeedOperator][Apply] No restriction, WEIGHT");
+      } else {
+        // Active
```
Comment (jeremylt) [inline backends/occa/ceed-occa-operator.c]: I also caught that the OCCA backend was not distinguishing between active and passive inputs with 'CEED_RESTRICTION_IDENTITY'.

File: backends/magma/ceed-magma.c [hunk 2]
```diff
@@ -682,7 +682,7 @@ static int CeedBasisDestroy_Magma(CeedBasis basis) {
 static int CeedBasisCreateTensorH1_Magma(Ceed ceed, CeedInt dim, CeedInt P1d,
     CeedInt Q1d, const CeedScalar *interp1d,
     const CeedScalar *grad1d,
-    const CeedScalar *qref1d,
+    const CeedScalar *qmagma1d,
```
Comment (jedbrown) [inline backends/magma/ceed-magma.c]: I don't think this is correct. "qref" is referring to the location of quadrature points on the reference element, not the `ref` backend.

File: backends/magma/ceed-magma.c [hunk 3]
```diff
@@ -798,21 +798,19 @@ static int CeedOperatorSetup_Magma(CeedOperator op) {
...
   for (CeedInt i=0; i<qf->numinputfields; i++) {
     CeedEvalMode emode = qf->inputfields[i].emode;
-    opmagma->numqin += !!(emode & CEED_EVAL_INTERP) + !!(emode & CEED_EVAL_GRAD) +
-                       !!
-                       (emode & CEED_EVAL_WEIGHT);
+    opmagma->numqin += !!(emode & CEED_EVAL_INTERP) + !!(emode & CEED_EVAL_GRAD) + !!
+                     (emode & CEED_EVAL_WEIGHT);
     opmagma->numein +=
-      !!op->inputfields[i].Erestrict; // Need E-vector when restriction exists
+      !!(op->inputfields[i].Erestrict != CEED_RESTRICTION_IDENTITY); // Need E-vector when non-identity restriction exists
```
Comment (jedbrown) [inline backends/magma/ceed-magma.c]: The `!!` (collapse nonzero values to 1) is not needed now.

File: backends/magma/ceed-magma.c [hunk 4]
```diff
@@ -798,21 +798,19 @@ static int CeedOperatorSetup_Magma(CeedOperator op) {
...
   }
   for (CeedInt i=0; i<qf->numoutputfields; i++) {
     CeedEvalMode emode = qf->outputfields[i].emode;
     opmagma->numqout += !!(emode & CEED_EVAL_INTERP) + !!(emode & CEED_EVAL_GRAD);
-    opmagma->numeout += !!op->outputfields[i].Erestrict;
+    opmagma->numeout += !!(op->outputfields[i].Erestrict != CEED_RESTRICTION_IDENTITY);
```
Comment (jedbrown) [inline backends/magma/ceed-magma.c]: Or here.

File: backends/magma/ceed-magma.c (hunk 2 above)
Comment (jeremylt) [reply backends/magma/ceed-magma.c]: Fixed

**Pull request merged**